packages/kbn-test/src/functional_tests/lib/run_elasticsearch.js

```diff
@@ -0,0 +1,40 @@
+import fs from 'fs';
+import { resolve } from 'path';
+
+import { log } from './log';
+import { extractTarball } from './tarball';
+import { findMostRecentlyChanged } from './find_most_recently_changed';
+import {
+  RELATIVE_ES_BIN,
+  ES_GRADLE_WRAPPER_BIN,
+  ES_ARCHIVE_PATTERN,
+  ES_REPO_ROOT,
+} from './paths';
+
+async function setupElasticsearch({ esExtractPath, procs }) {
+  await procs.run('buildEs', {
+    cmd: ES_GRADLE_WRAPPER_BIN,
+    args: [':distribution:archives:tar:assemble'],
+    cwd: ES_REPO_ROOT,
+    wait: true,
+  });
+
+  const esTarballPath = findMostRecentlyChanged(ES_ARCHIVE_PATTERN);
+  log.debug('es build output %j', esTarballPath);
+
+  await extractTarball(esTarballPath, esExtractPath);
+}
+
+export async function runElasticsearch({ tmpDir, procs, config }) {
```

Comment:
This should be replaced with using `kbn/es`, and doing so the following files can be removed:

- packages/kbn-test/src/functional_tests/lib/find_most_recently_changed.js
- packages/kbn-test/src/functional_tests/lib/tarball.js
- x-pack/dev-tools/functional_tests/lib/tarball.js
- x-pack/dev-tools/functional_tests/lib/find_most_recently_changed.js

And some paths in `packages/kbn-test/src/functional_tests/lib/paths.js` and `x-pack/dev-tools/functional_tests/lib/paths.js` can also be removed

Reply:
> Added comments inline which should allow you to entirely get rid of runEsWithXpack.

> This should be replaced with using kbn/es,

So, you're saying we get rid of `runElasticsearch` and `runEsWithXpack`, both?

Reply:
^ Nevermind, yes, that makes sense. Both would go away, or just be simple methods that use kbn/es